deepchem/models/torch_models/infograph.py

```diff
@@ -0,0 +1,429 @@
+            'fc1':
+                torch.nn.Linear(2 * self.embedding_dim, self.embedding_dim),
+            'fc2':
+                torch.nn.Linear(self.embedding_dim, 1),
+            'local_d':
+                MultilayerPerceptron(self.embedding_dim,
+                                     self.embedding_dim, (self.embedding_dim,),
+                                     skip_connection=True),
+            'global_d':
+                MultilayerPerceptron(2 * self.embedding_dim,
+                                     self.embedding_dim, (self.embedding_dim,),
+                                     skip_connection=True),
+        }
+
+    def build_model(self):
+        return InfoGraph(**self.components)
+
+    def loss_func(self, inputs, labels, weights):
+        if self.use_unsup_loss:
+            sup_loss = F.mse_loss(self.model(inputs), labels)
+            unsup_loss = self.unsup_loss(inputs)
+            if self.separate_encoder:
+                unsup_sup_loss = self.unsup_sup_loss(inputs, labels, weights)
+                loss = sup_loss + unsup_loss + unsup_sup_loss * self.learning_rate
+            else:
+                loss = sup_loss + unsup_loss * self.learning_rate
+            return (loss * weights).mean()
+        else:
+            sup_loss = F.mse_loss(self.model(inputs), labels)
+            return (sup_loss * weights).mean()
+
+    def unsup_loss(self, inputs):
+        if self.separate_encoder:
+            y, M = self.components['unsup_encoder'](inputs)
+        else:
+            y, M = self.components['encoder'](inputs)
+        g_enc = self.components['global_d'](y)
+        l_enc = self.components['local_d'](M)
+
+        if self.local:
+            loss = self.local_global_loss_(l_enc, g_enc, inputs.graph_index)
+        return loss
+
+    def _prepare_batch(self, batch):
+        inputs, labels, weights = batch
+        inputs = BatchGraphData(inputs[0])
+        inputs.edge_features = torch.from_numpy(
+            inputs.edge_features).float().to(self.device)
+        inputs.edge_index = torch.from_numpy(inputs.edge_index).long().to(
+            self.device)
+        inputs.node_features = torch.from_numpy(
+            inputs.node_features).float().to(self.device)
+        inputs.graph_index = torch.from_numpy(inputs.graph_index).long().to(
+            self.device)
+
+        _, labels, weights = super()._prepare_batch(([], labels, weights))
+
+        if (len(labels) != 0) and (len(weights) != 0):
+            labels = labels[0]
+            weights = weights[0]
+
+        return inputs, labels, weights
+
+    def unsup_sup_loss(self, inputs, labels, weights):
+        y, M = self.components['encoder'](inputs)
+        y_, M_ = self.components['unsup_encoder'](inputs)
+
+        g_enc = self.components['ff1'](y)
+        g_enc1 = self.components['ff2'](y_)
+
+        loss = self.global_global_loss_(g_enc, g_enc1)
+        return loss
+
+    def local_global_loss_(self, l_enc, g_enc, batch):
```

Comment:
Do you want to do `_local_global_loss` to mark this as private? I think the underscore should come up front